### PR TITLE
WIP: Kill all globals in rabin2.c

### PR DESCRIPTION
**Pull request by radare**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](../DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](../DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](../../radare2book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

**Comment by XVilka** [comment]
It breaks rabin2 tests:
https://travis-ci.com/radareorg/radare2/jobs/287908832#L687
```
[XX] db/tools/rabin2 rabin2 -s file elf  325
$ r2 -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Q -i /tmp/tmp-17108PLbrLsB4pPiG ../bins/elf/analysis/x86-helloworld-gcc
!rabin2 -s ${R2_FILE}|wc -l|awk "{print \$1}"
-69
+3
EXPECT='3'
[XX] db/tools/rabin2 rabin2 -S, -SS, -rS and -rSS file elf  717
$ r2 -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Q -i /tmp/tmp-17108irALweqRZXVm ../bins/elf/analysis/x86-helloworld-gcc
!rabin2 -S ${R2_FILE}|wc -l|awk "{print \$1}"
!rabin2 -SS ${R2_FILE}|wc -l|awk "{print \$1}"
!env RABIN2_PREFIX=prixfixe rabin2 -S ${R2_FILE} |grep prixfixe. |wc -l |awk "{print \$1}"
!env RABIN2_PREFIX=prixfixe rabin2 -SS ${R2_FILE} |grep prixfixe. |wc -l |awk "{print \$1}"
-35
-14
-30
-9
+4
+4
+0
+0
EXPECT='4
4
0
0'
```

**Comment by ret2libc** [comment]
Let's mark this WIP until tests are red.

**Comment by radare** [comment]
Until -> while

**Comment by ret2libc** [comment]
Woops, right :D

File: libr/main/rabin2.c [hunk 1]
```diff
@@ -100,8 +90,14 @@ static int rabin_show_help(int v) {
 	return 1;
 }
 
-static char *stdin_gets() {
+static char *stdin_gets(bool liberate) {
+	static char *stdin_buf = NULL;
```
Comment (ret2libc) [inline libr/main/rabin2.c]: why static?

**Pull request merged**